backend/tests/e2e/test_api_e2e.py

```
@@ -0,0 +1,988 @@
+                        # Patch the semaphore directly in the main module, not just the return value
+                        import tarsy.main
+                        tarsy.main.alert_processing_semaphore = real_semaphore
+                        
+                        # CRITICAL: Also patch the mock to return the real semaphore when needed
+                        mock_semaphore.return_value = real_semaphore
+                        
+                        # Initialize database
+                        from tarsy.database.init_db import initialize_database
+                        db_success = initialize_database()
+                        if not db_success:
+                            assert False, "Database initialization failed"
```

Ruff (0.12.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)